src/Knp/Snappy/AbstractGenerator.php

```diff
@@ -571,21 +576,15 @@ protected function isAssociativeArray(array $array): bool
     }
 
     /**
-     * Executes the given command via shell and returns the complete output as
-     * a string.
+     * Executes the given process and returns the complete output as a string.
      *
-     * @param string $command
+     * @param Process  $process
+     * @param string[] $command Basically just for testing purposes.
      *
      * @return array [status, stdout, stderr]
      */
-    protected function executeCommand(string $command): array
+    protected function runProcess(Process $process, array $command): array
```

Comment:
Do we really need the $command parameter here?
I'm not so sure I understand why it's needed for test purposes.

I think we can find a way to refactoring the Pdf test case in order to avoid having to add this parameter.
I can work on it if you prefer.

Reply:
I kept the commit history so you can see (right now, in the most current commit) how tests looked like without this parameter. And of course I'm 👍 for getting rid of it, but have no clue.

Reply:
Ok, I'm gonna be off for the next couple weeks but I'm gonna have a look at it when I'll be back :wink:

Reply:
Ha, enjoy! 🌞

Reply:
Sorry for the delay, I have been very busy lately.
If adding the `$command` option to this function it's just needed to be able to assert the last executed command in `tests/Knp/Snappy/PdfTest.php` test, than you can remove it and change the `PDFSpy` test class from:
```
class PdfSpy extends Pdf
{
    ...
    protected function runProcess(Process $process, array $command): array
    {
        $this->lastCommand = $command;

        return [0, 'output', 'errorOutput'];
    }
    ...
}
```
to:
```
class PdfSpy extends Pdf
{
    ...
    public function getCommand($input, string $output, array $options = []): array
    {
        $this->lastCommand = parent::getCommand($input, $output, $options);

        return $this->lastCommand;
    }

    protected function runProcess(Process $process): array
    {
        return [0, 'output', 'errorOutput'];
    }
    ...
}
```
It should do the trick.

A test refactoring is needed tho :thinking: but it's out of the scope of this PR :wink:

Reply:
> It should do the trick.

Alright, done.